libs/blocks/merch-card/legacy-merch-card.js

```diff
@@ -0,0 +1,185 @@
+  'special-offers': SPECIAL_OFFERS,
+  plans: PLANS_CARD,
+};
+const getPodType = (styles) => {
+  const authoredType = styles?.find((style) => style in cardTypes);
+  return cardTypes[authoredType] || SEGMENT_BLADE;
+};
+
+const decorateFooter = (el, altCtaMetaData, cardType) => {
+  const cardFooter = el.querySelector('.consonant-CardFooter');
+  const decorateWithSecureTransactionSign = () => {
+    const secureTransactionWrapper = createTag('div', { class: 'secure-transaction-wrapper' });
+    const label = createTag('span', { class: 'secure-transaction-label' });
+    const secureElement = createTag('span', { class: 'secure-transaction-icon' });
+    secureTransactionWrapper.append(secureElement, label);
+    replaceKey('secure-transaction', getConfig()).then((replacedKey) => {
+      label.textContent = replacedKey;
+    });
+    return secureTransactionWrapper;
+  };
+
+  const createCheckbox = (checkBoxText) => {
+    const container = createTag('label', { class: 'checkbox-container' });
+    const input = createTag('input', { id: 'alt-cta', type: 'checkbox' });
+    const checkmark = createTag('span', { class: 'checkmark' });
+    const label = createTag('span', { class: 'checkbox-label' }, checkBoxText.innerHTML);
+    container.append(input, checkmark, label);
+    return container;
+  };
+
+  const createSecureSign = () => {
+    const cardFooterRow = el.querySelector('.consonant-CardFooter-row');
+    const standardWrapper = createTag('div', { class: 'standard-wrapper' });
+    const secureTransactionWrapper = decorateWithSecureTransactionSign();
+    standardWrapper.append(secureTransactionWrapper, cardFooterRow);
+    cardFooter?.append(standardWrapper);
+  };
+
+  const decorateAlternativeCta = () => {
+    const altCtaRegex = /href=".*"/;
+    if (!altCtaRegex.test(altCtaMetaData[1]?.innerHTML)) return;
+    const cardFooterRow = el.querySelector('.consonant-CardFooter-row');
+    const originalCtaButton = cardFooterRow.querySelector('.consonant-CardFooter-cell--right');
+    const checkboxContainer = createCheckbox(altCtaMetaData[0]);
+    const altCtaButtonData = altCtaMetaData[1];
+    decorateButtons(altCtaButtonData);
+    const altCtaButton = createTag('div', { class: originalCtaButton.classList }, altCtaButtonData.innerHTML);
+    altCtaButton.classList.add('button--inactive');
+    checkboxContainer.querySelector('input[type="checkbox"]').addEventListener('change', ({ target: { checked } }) => {
+      originalCtaButton.classList.toggle('button--inactive', checked);
+      altCtaButton.classList.toggle('button--inactive', !checked);
+    });
+    cardFooterRow.append(altCtaButton);
+    cardFooter.prepend(checkboxContainer);
+    altCtaMetaData[0].parentNode.remove();
+  };
+  if (altCtaMetaData !== null) decorateAlternativeCta();
+  if (el.classList.contains('secure')) createSecureSign();
+  cardFooter.querySelectorAll('.consonant-CardFooter-cell').forEach((cell) => cell.classList.add(`consonant-${cardType}-cell`));
```

Comment:
not sure why linter didn't flag this - line length > 100.  There's actually quite a few in this file.

Reply:
fixed